app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/DefaultTaskRepository.kt

```diff
@@ -18,25 +18,32 @@ package com.example.android.architecture.blueprints.todoapp.data
 
 import com.example.android.architecture.blueprints.todoapp.data.source.local.TaskDao
 import com.example.android.architecture.blueprints.todoapp.data.source.network.NetworkDataSource
+import javax.inject.Inject
+import javax.inject.Singleton
 import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 /**
  * Default implementation of [TaskRepository]. Single entry point for managing tasks' data.
  *
  * @param tasksNetworkDataSource - The network data source
  * @param taskDao - The local data source
- * @param coroutineDispatcher - The dispatcher to be used for long running or complex operations,
- * such as network calls or mapping many models. This is important to avoid blocking the calling
- * thread.
+ * @param expensiveWorkDispatcher - The dispatcher to be used for long running or complex
+ * operations, such as ID generation or mapping many models. This is important to avoid blocking the
+ * calling thread.
+ * @param fireAndForgetScope - The coroutine scope used for fire-and-forget jobs, such as sending
+ * data to the network.
  */
-class DefaultTaskRepository(
+@Singleton
+class DefaultTaskRepository @Inject constructor(
     private val tasksNetworkDataSource: NetworkDataSource,
     private val taskDao: TaskDao,
-    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.Default
+    private val expensiveWorkDispatcher: CoroutineDispatcher,
+    private val fireAndForgetScope: CoroutineScope,
```

Comment:
```suggestion
    private val scope: CoroutineScope,
```

All scopes are fire and forget. It's fine to say `externalScope` or simply `scope`